### PR TITLE
[WIP] Add detectableSwipeDirections

### PR DESCRIPTION
**Pull request by HeavenOSK**
# Overview 
The another solution of #41 .

### CODE REVIEW

File: lib/src/model/swipe_rate_per_threshold.dart [hunk 1]
```diff
@@ -10,41 +10,102 @@ class _SwipeRatePerThreshold {
...
+      horizontalRatePerThreshold(),
+    ].whereType<_SwipeRatePerThreshold>().toList();
+    if (rateList.isEmpty) {
+      return null;
+    }
+    return (rateList
+          ..sort(
+            (a, b) => a.rate > b.rate ? -1 : 1,
+          ))
+        .first;
```
Comment (MaddinMade) [inline lib/src/model/swipe_rate_per_threshold.dart:91]:
```suggestion
    return rateList.length == 1 || rateList[0].rate > rateList[1] ? rateList[0] : rateList[1];
```
I don't know if the sort function is a little overkill at this point. But it is of course more elegant and flexible than the brute force method.

**Review by MaddinMade — APPROVED** [review]
The changes are definitely more user-friendly than my approach and well implemented.
Nice work :)

**Comment by MaddinMade** [comment]
I guess I was trying to cover use cases that either never occur or occur far too infrequently and neglected user experience.

File: lib/src/model/swipe_rate_per_threshold.dart (hunk 1 above)
Comment (HeavenOSK) [reply lib/src/model/swipe_rate_per_threshold.dart]: Thanks for your suggestion. This code is more understandable! 🙏

**Comment by HeavenOSK** [comment]
@MaddinMade 
Thank you for reviewing & your contribution! 🙏 
> I guess I was trying to cover use cases that either never occur or occur far too infrequently and neglected user experience.
 
It's OK! 👍  I think it's hard to judge correctly also to me.
 
So, I'll merge this PR and will release soon!

**Pull request merged**